### PR TITLE
Add {de,}serialization for signed ints

### PR DESCRIPTION
**Pull request by Shimuuar**
Parsers for signed ints are useful for parsing existing data formats (I've defined similar functions in the past). Encoders are useful too as shown by #33.

### CODE REVIEW

**Comment by Shimuuar** [comment]
Is this PR going to be merged in 0.8.0.0?

File: src/Data/Binary/Get.hs [hunk 1]
```diff
@@ -427,6 +446,12 @@ getWord8 :: Get Word8
 getWord8 = readN 1 B.unsafeHead
 {-# INLINE getWord8 #-}
 
+-- | Read Int8 from the monad state
```
Comment (kolmodin) [inline src/Data/Binary/Get.hs]: "Read an Int8 from the monad state"

File: src/Data/Binary/Get.hs [hunk 2]
```diff
@@ -520,6 +545,39 @@ word64le = \s ->
 {-# INLINE getWord64le #-}
 {-# INLINE word64le #-}
 
+
+-- | Read a Int16 in big endian format
+getInt16be :: Get Int16
+getInt16be = fromIntegral <$> getWord16be
+{-# INLINE getInt16be #-}
+
+-- | Read a Int16 in big endian format
```
Comment (kolmodin) [inline src/Data/Binary/Get.hs]: Int32

File: src/Data/Binary/Get.hs [hunk 3]
```diff
@@ -520,6 +545,39 @@ word64le = \s ->
...
+getInt32be :: Get Int32
+getInt32be =  fromIntegral <$> getWord32be
+{-# INLINE getInt32be #-}
+
+-- | Read a Int16 in big endian format
```
Comment (kolmodin) [inline src/Data/Binary/Get.hs]: an Int64

File: src/Data/Binary/Get.hs [hunk 4]
```diff
@@ -520,6 +545,39 @@ word64le = \s ->
...
+getInt64be :: Get Int64
+getInt64be = fromIntegral <$> getWord64be
+{-# INLINE getInt64be #-}
+
+
+-- | Read a Int16 in little endian format
```
Comment (kolmodin) [inline src/Data/Binary/Get.hs]: an Int16

File: src/Data/Binary/Get.hs [hunk 5]
```diff
@@ -520,6 +545,39 @@ word64le = \s ->
...
+getInt16le :: Get Int16
+getInt16le = fromIntegral <$> getWord16le
+{-# INLINE getInt16le #-}
+
+-- | Read a Int32 in little endian format
```
Comment (kolmodin) [inline src/Data/Binary/Get.hs]: an Int32

File: src/Data/Binary/Get.hs [hunk 6]
```diff
@@ -520,6 +545,39 @@ word64le = \s ->
...
+getInt32le :: Get Int32
+getInt32le =  fromIntegral <$> getWord32le
+{-# INLINE getInt32le #-}
+
+-- | Read a Int64 in little endian format
```
Comment (kolmodin) [inline src/Data/Binary/Get.hs]: an Int64

**Comment by Shimuuar** [comment]
I fixed documentation issues.

**Comment by kolmodin** [comment]
The build is failing on travis for < GHC 7.10. See https://travis-ci.org/kolmodin/binary/builds/90705106
You need to conditionally import Control.Applicative.
For an example, see https://github.com/kolmodin/binary/commit/67ccd5398f199f3e92cc7c34dcf0fa47d41c3011

**Comment by Shimuuar** [comment]
Fixed.

**Comment by Shimuuar** [comment]
Ping? Any update on this PR?

**Comment by kolmodin** [comment]
Thanks for the PR, and for addressing all comments :)

**Pull request merged**